walk-in-interview/src/main/java/com/google/job/data/JobsDatabase.java

```diff
@@ -0,0 +1,52 @@
+package com.google.job.data;
+
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+
+import java.util.concurrent.Future;
+
+/** Helps persist and retrieve Future of job posts. */
+public final class JobsDatabase {
+    private static final String JOB_COLLECTION = "Jobs";
+    private final Firestore firestore;
+
+    public JobsDatabase(Firestore firestore) {
+        this.firestore = firestore;
+    }
+
+    /**
+     * Adds a newly created job post.
+     *
+     * @param newJob Newly created job post. Assumes that it is non-nullable.
+     * @return Future for the added job post.
+     */
+    public Future<DocumentReference> addJob(Job newJob) {
```

Comment:
Btw, can the future be ```Future<Job>``` instead? So the servlet only work with job objects instead of fire store documents?
You can try Futures.transform
https://guava.dev/releases/19.0/api/docs/com/google/common/util/concurrent/Futures.html#transform(com.google.common.util.concurrent.ListenableFuture,%20com.google.common.base.Function)